parallel-install/pkg/preinstaller/preinstaller.go

```diff
@@ -0,0 +1,172 @@
+package preinstaller
+
+import (
+	"fmt"
+	"github.com/avast/retry-go"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/config"
+	"io/ioutil"
+	"k8s.io/client-go/dynamic"
+	"os"
+)
+
+// PreInstaller prepares k8s cluster for Kyma installation.
+type PreInstaller struct {
+	applier       ResourceApplier
+	cfg           config.Config
+	dynamicClient dynamic.Interface
+	retryOptions  []retry.Option
+}
+
+// File consists of a path to the file that was a part of PreInstaller installation
+// and a component name that it belongs to.
+type File struct {
+	component string
+	path      string
+}
+
+// Output contains lists of installed and not installed files during PreInstaller installation.
+type Output struct {
+	installed    []File
+	notInstalled []File
+}
+
+type resourceInfoInput struct {
+	dirSuffix                string
+	resourceType             string
+	installationResourcePath string
+}
+
+type resourceInfoResult struct {
+	component    string
+	name         string
```

Comment:
I understand that this holds the file name? I'd rename it to `fileName` then. Just the "name" requires looking at the code to realize "name of what" is stored there.